// @storybook/preset-react-webpack/dist/framework-preset-react-docs.js
import CJS_COMPAT_NODE_URL_gcrvfg3p0j from 'node:url';
import CJS_COMPAT_NODE_PATH_gcrvfg3p0j from 'node:path';
import CJS_COMPAT_NODE_MODULE_gcrvfg3p0j from "node:module";

var __filename = CJS_COMPAT_NODE_URL_gcrvfg3p0j.fileURLToPath(import.meta.url);
var __dirname = CJS_COMPAT_NODE_PATH_gcrvfg3p0j.dirname(__filename);
var require = CJS_COMPAT_NODE_MODULE_gcrvfg3p0j.createRequire(import.meta.url);

// ------------------------------------------------------------
// end of CJS compatibility banner, injected by Storybook's esbuild configuration
// ------------------------------------------------------------

// src/framework-preset-react-docs.ts
import { fileURLToPath } from "node:url";
var webpackFinal = async (config, options) => {
  let typescriptOptions = await options.presets.apply("typescript", {}), debug = options.loglevel === "debug", { reactDocgen, reactDocgenTypescriptOptions } = typescriptOptions || {};
  if (typeof reactDocgen != "string")
    return config;
  if (reactDocgen !== "react-docgen-typescript")
    return {
      ...config,
      module: {
        ...config.module ?? {},
        rules: [
          ...config.module?.rules ?? [],
          {
            test: /\.(cjs|mjs|tsx?|jsx?)$/,
            enforce: "pre",
            loader: fileURLToPath(
              import.meta.resolve("@storybook/preset-react-webpack/react-docgen-loader")
            ),
            options: {
              debug
            },
            exclude: /(\.(stories|story)\.(js|jsx|ts|tsx))|(node_modules)/
          }
        ]
      }
    };
  let { ReactDocgenTypeScriptPlugin } = await import("@storybook/react-docgen-typescript-plugin");
  return {
    ...config,
    module: {
      ...config.module ?? {},
      rules: [
        ...config.module?.rules ?? [],
        {
          test: /\.(cjs|mjs|jsx?)$/,
          enforce: "pre",
          loader: fileURLToPath(
            import.meta.resolve("@storybook/preset-react-webpack/react-docgen-loader")
          ),
          options: {
            debug
          },
          exclude: /(\.(stories|story)\.(js|jsx|ts|tsx))|(node_modules)/
        }
      ]
    },
    plugins: [
      ...config.plugins || [],
      new ReactDocgenTypeScriptPlugin({
        ...reactDocgenTypescriptOptions,
        // We *need* this set so that RDT returns default values in the same format as react-docgen
        savePropValueAsString: !0
      })
    ]
  };
};
export {
  webpackFinal
};
