// @storybook/nextjs/dist/preset.js
import CJS_COMPAT_NODE_URL_ii3a4tu75d7 from 'node:url';
import CJS_COMPAT_NODE_PATH_ii3a4tu75d7 from 'node:path';
import CJS_COMPAT_NODE_MODULE_ii3a4tu75d7 from "node:module";

var __filename = CJS_COMPAT_NODE_URL_ii3a4tu75d7.fileURLToPath(import.meta.url);
var __dirname = CJS_COMPAT_NODE_PATH_ii3a4tu75d7.dirname(__filename);
var require = CJS_COMPAT_NODE_MODULE_ii3a4tu75d7.createRequire(import.meta.url);

// ------------------------------------------------------------
// end of CJS compatibility banner, injected by Storybook's esbuild configuration
// ------------------------------------------------------------
import {
  addScopedAlias,
  isNextVersionGte,
  resolveNextConfig
} from "./_node-chunks/chunk-HTZOSU3S.js";
import "./_node-chunks/chunk-4TF7YMRH.js";

// src/preset.ts
import { existsSync } from "node:fs";
import { join } from "node:path";
import { fileURLToPath as fileURLToPath2 } from "node:url";
import { getProjectRoot } from "storybook/internal/common";
import { logger } from "storybook/internal/node-logger";
import { loadPartialConfig } from "@babel/core";
import semver from "semver";

// src/babel/preset.ts
import { createRequire } from "node:module";
import { dirname } from "node:path";

// src/babel/plugins/amp-attributes.ts
function AmpAttributePatcher() {
  return {
    visitor: {
      JSXOpeningElement(path) {
        let openingElement = path.node, { name, attributes } = openingElement;
        if (name && name.type === "JSXIdentifier" && name.name.startsWith("amp-"))
          for (let attribute of attributes)
            attribute.type === "JSXAttribute" && attribute.name.name === "className" && (attribute.name.name = "class");
      }
    }
  };
}

// src/babel/plugins/jsx-pragma.ts
import jsx from "next/dist/compiled/babel/plugin-syntax-jsx.js";
function jsxPragma({ types: t }) {
  return {
    inherits: jsx,
    visitor: {
      JSXElement(_path, state) {
        state.set("jsx", !0);
      },
      // Fragment syntax is still JSX since it compiles to createElement(),
      // but JSXFragment is not a JSXElement
      JSXFragment(_path, state) {
        state.set("jsx", !0);
      },
      Program: {
        exit(path, state) {
          if (state.get("jsx")) {
            let pragma = t.identifier(state.opts.pragma), importAs = pragma, existingBinding = state.opts.reuseImport !== !1 && state.opts.importAs && path.scope.getBinding(state.opts.importAs);
            if (state.opts.property) {
              state.opts.importAs ? importAs = t.identifier(state.opts.importAs) : importAs = path.scope.generateUidIdentifier("pragma");
              let mapping = t.variableDeclaration("var", [
                t.variableDeclarator(
                  pragma,
                  t.memberExpression(importAs, t.identifier(state.opts.property))
                )
              ]), newPath;
              existingBinding && t.isVariableDeclarator(existingBinding.path.node) && t.isCallExpression(existingBinding.path.node.init) && t.isIdentifier(existingBinding.path.node.init.callee) && existingBinding.path.node.init.callee.name === "require" ? [newPath] = existingBinding.path.parentPath.insertAfter(mapping) : [newPath] = path.unshiftContainer("body", mapping);
              for (let declar of newPath.get("declarations")) {
                let kind = ["var", "let", "const"].find((k) => newPath.node.kind === k) || "const";
                path.scope.registerBinding(kind, declar);
              }
            }
            if (!existingBinding) {
              let importSpecifier = t.importDeclaration(
                [
                  state.opts.import ? (
                    // import { $import as _pragma } from '$module'
                    t.importSpecifier(importAs, t.identifier(state.opts.import))
                  ) : state.opts.importNamespace ? t.importNamespaceSpecifier(importAs) : (
                    // import _pragma from '$module'
                    t.importDefaultSpecifier(importAs)
                  )
                ],
                t.stringLiteral(state.opts.module || "react")
              ), [newPath] = path.unshiftContainer("body", importSpecifier);
              for (let specifier of newPath.get("specifiers"))
                path.scope.registerBinding("module", specifier);
            }
          }
        }
      }
    }
  };
}

// src/babel/plugins/optimize-hook-destructuring.ts
var isHook = /^use[A-Z]/, isBuiltInHook = /^use(Callback|Context|DebugValue|Effect|ImperativeHandle|LayoutEffect|Memo|Reducer|Ref|State)$/;
function optimizeHookDestructuring({
  types: t
}) {
  let visitor = {
    CallExpression(path, state) {
      let { onlyBuiltIns } = state.opts, libs = state.opts.lib && (state.opts.lib === !0 ? ["react", "preact/hooks"] : [].concat(state.opts.lib));
      if (!t.isVariableDeclarator(path.parent) || !t.isArrayPattern(path.parent.id))
        return;
      let hookName = path.node.callee.name;
      if (libs) {
        let binding = path.scope.getBinding(hookName);
        if (!binding || binding.kind !== "module")
          return;
        let specifier = binding.path.parent.source.value;
        if (!libs.some((lib) => lib === specifier))
          return;
      }
      (onlyBuiltIns ? isBuiltInHook : isHook).test(hookName) && (path.parent.id = t.objectPattern(
        path.parent.id.elements.reduce((patterns, element, i) => element === null ? patterns : patterns.concat(t.objectProperty(t.numericLiteral(i), element)), [])
      ));
    }
  };
  return {
    name: "optimize-hook-destructuring",
    visitor: {
      // this is a workaround to run before preset-env destroys destructured assignments
      Program(path, state) {
        path.traverse(visitor, state);
      }
    }
  };
}

// src/babel/plugins/react-loadable-plugin.ts
import { relative as relativePath } from "node:path";
function react_loadable_plugin_default({ types: t }) {
  return {
    visitor: {
      ImportDeclaration(path, state) {
        if (path.node.source.value !== "next/dynamic")
          return;
        let defaultSpecifier = path.get("specifiers").find((specifier) => specifier.isImportDefaultSpecifier());
        if (!defaultSpecifier)
          return;
        let bindingName = defaultSpecifier.node.local.name, binding = path.scope.getBinding(bindingName);
        binding && binding.referencePaths.forEach((refPath) => {
          let callExpression = refPath.parentPath;
          if (callExpression.isMemberExpression() && callExpression.node.computed === !1) {
            let property = callExpression.get("property");
            !Array.isArray(property) && property.isIdentifier({ name: "Map" }) && (callExpression = callExpression.parentPath);
          }
          if (!callExpression.isCallExpression())
            return;
          let callExpression_ = callExpression, args = callExpression_.get("arguments");
          if (args.length > 2)
            throw callExpression_.buildCodeFrameError("next/dynamic only accepts 2 arguments");
          if (!args[0])
            return;
          let loader, options;
          if (args[0].isObjectExpression() ? options = args[0] : (args[1] || callExpression_.node.arguments.push(t.objectExpression([])), args = callExpression_.get("arguments"), loader = args[0], options = args[1]), !options.isObjectExpression())
            return;
          let properties = options.get("properties"), propertiesMap = {};
          if (properties.forEach((property) => {
            let key = property.get("key");
            propertiesMap[key.node.name] = property;
          }), propertiesMap.loadableGenerated || (propertiesMap.loader && (loader = propertiesMap.loader.get("value")), propertiesMap.modules && (loader = propertiesMap.modules.get("value")), !loader || Array.isArray(loader)))
            return;
          let dynamicImports = [], dynamicKeys = [];
          if (propertiesMap.ssr) {
            let ssr = propertiesMap.ssr.get("value"), nodePath = Array.isArray(ssr) ? void 0 : ssr;
            nodePath && nodePath.node.type === "BooleanLiteral" && nodePath.node.value === !1 && loader && state.file.opts.caller?.isServer && loader.replaceWith(t.arrowFunctionExpression([], t.nullLiteral(), !0));
          }
          if (loader.traverse({
            Import(importPath) {
              let importArguments = importPath.parentPath.get("arguments");
              if (!Array.isArray(importArguments))
                return;
              let { node } = importArguments[0];
              dynamicImports.push(node), dynamicKeys.push(
                t.binaryExpression(
                  "+",
                  t.stringLiteral(
                    `${state.file.opts.caller?.pagesDir ? relativePath(state.file.opts.caller.pagesDir, state.file.opts.filename) : state.file.opts.filename} -> `
                  ),
                  node
                )
              );
            }
          }), !!dynamicImports.length && (options.node.properties.push(
            t.objectProperty(
              t.identifier("loadableGenerated"),
              t.objectExpression(
                state.file.opts.caller?.isDev || state.file.opts.caller?.isServer ? [t.objectProperty(t.identifier("modules"), t.arrayExpression(dynamicKeys))] : [
                  t.objectProperty(
                    t.identifier("webpack"),
                    t.arrowFunctionExpression(
                      [],
                      t.arrayExpression(
                        dynamicImports.map((dynamicImport) => t.callExpression(
                          t.memberExpression(
                            t.identifier("require"),
                            t.identifier("resolveWeak")
                          ),
                          [dynamicImport]
                        ))
                      )
                    )
                  )
                ]
              )
            )
          ), loader.isCallExpression())) {
            let arrowFunction = t.arrowFunctionExpression([], loader.node);
            loader.replaceWith(arrowFunction);
          }
        });
      }
    }
  };
}

// src/babel/preset.ts
var isLoadIntentTest = process.env.NODE_ENV === "test", isLoadIntentDevelopment = process.env.NODE_ENV === "development", require2 = createRequire(import.meta.url);
function styledJsxOptions(options) {
  return options = options || {}, options.styleModule = "styled-jsx/style", Array.isArray(options.plugins) && (options.plugins = options.plugins.map((plugin) => {
    if (Array.isArray(plugin)) {
      let [name, pluginOptions] = plugin;
      return [require2.resolve(name), pluginOptions];
    }
    return require2.resolve(plugin);
  })), options;
}
function supportsStaticESM(caller) {
  return !!caller?.supportsStaticESM;
}
var preset_default = (api, options = {}) => {
  let supportsESM = api.caller(supportsStaticESM), isServer = api.caller((caller) => !!caller && caller.isServer), isCallerDevelopment = api.caller((caller) => caller?.isDev), isTest = isCallerDevelopment == null && isLoadIntentTest, isDevelopment = isCallerDevelopment === !0 || isCallerDevelopment == null && isLoadIntentDevelopment, isBabelLoader = api.caller(
    (caller) => !!caller && (caller.name === "babel-loader" || caller.name === "next-babel-turbo-loader")
  ), useJsxRuntime = options["preset-react"]?.runtime === "automatic" || !!api.caller((caller) => !!caller && caller.hasJsxRuntime) && options["preset-react"]?.runtime !== "classic", presetEnvConfig = {
    // In the test environment `modules` is often needed to be set to true, babel figures that out by itself using the `'auto'` option
    // In production/development this option is set to `false` so that webpack can handle import/export with tree-shaking
    modules: "auto",
    exclude: ["transform-typeof-symbol"],
    bugfixes: !0,
    targets: {
      chrome: 100,
      safari: 15,
      firefox: 91
    },
    ...options["preset-env"]
  };
  return (isServer || isTest) && (!presetEnvConfig.targets || !(typeof presetEnvConfig.targets == "object" && "node" in presetEnvConfig.targets)) && (presetEnvConfig.targets = {
    // Targets the current process' version of Node. This requires apps be
    // built and deployed on the same version of Node.
    // This is the same as using "current" but explicit
    node: process.versions.node
  }), {
    sourceType: "unambiguous",
    presets: [
      [require2("@babel/preset-env"), presetEnvConfig],
      [
        require2("@babel/preset-react"),
        {
          // This adds @babel/plugin-transform-react-jsx-source and
          // @babel/plugin-transform-react-jsx-self automatically in development
          development: isDevelopment || isTest,
          ...useJsxRuntime ? { runtime: "automatic" } : { pragma: "__jsx" },
          ...options["preset-react"]
        }
      ],
      [
        require2("@babel/preset-typescript"),
        { allowNamespaces: !0, ...options["preset-typescript"] }
      ]
    ],
    plugins: [
      isDevelopment && require2.resolve("react-refresh/babel"),
      !useJsxRuntime && [
        jsxPragma,
        {
          // This produces the following injected import for modules containing JSX:
          //   import React from 'react';
          //   var __jsx = React.createElement;
          module: "react",
          importAs: "React",
          pragma: "__jsx",
          property: "createElement"
        }
      ],
      [
        optimizeHookDestructuring,
        {
          // only optimize hook functions imported from React/Preact
          lib: !0
        }
      ],
      require2("@babel/plugin-syntax-dynamic-import"),
      require2("@babel/plugin-syntax-import-assertions"),
      react_loadable_plugin_default,
      [require2("@babel/plugin-transform-class-properties"), options["class-properties"] || {}],
      [
        require2("@babel/plugin-transform-object-rest-spread"),
        {
          useBuiltIns: !0
        }
      ],
      !isServer && [
        require2("@babel/plugin-transform-runtime"),
        {
          corejs: !1,
          helpers: !0,
          regenerator: !0,
          useESModules: supportsESM && presetEnvConfig.modules !== "commonjs",
          absoluteRuntime: isBabelLoader ? dirname(require2.resolve("@babel/runtime/package.json")) : void 0,
          ...options["transform-runtime"]
        }
      ],
      [
        isTest && options["styled-jsx"] && options["styled-jsx"]["babel-test"] ? require2("styled-jsx/babel-test") : require2("styled-jsx/babel"),
        styledJsxOptions(options["styled-jsx"])
      ],
      AmpAttributePatcher,
      isServer && require2("@babel/plugin-syntax-bigint"),
      // Always compile numeric separator because the resulting number is
      // smaller.
      require2("@babel/plugin-transform-numeric-separator"),
      require2("@babel/plugin-transform-export-namespace-from")
    ].filter(Boolean)
  };
};

// src/config/webpack.ts
import { fileURLToPath } from "node:url";
var isNext16orNewer = isNextVersionGte("16.0.0"), tryResolve = (path) => {
  try {
    return fileURLToPath(import.meta.resolve(path)) || !1;
  } catch {
    return !1;
  }
}, configureConfig = async ({
  baseConfig,
  nextConfigPath
}) => {
  let nextConfig = await resolveNextConfig({ nextConfigPath });
  return isNext16orNewer || addScopedAlias(baseConfig, "next/config"), baseConfig.resolve?.alias?.["react-dom"] && delete baseConfig.resolve.alias?.["react-dom"], tryResolve("next/dist/compiled/react") && addScopedAlias(baseConfig, "react", "next/dist/compiled/react"), tryResolve("next/dist/compiled/react-dom/cjs/react-dom-test-utils.production.js") && addScopedAlias(
    baseConfig,
    "react-dom/test-utils",
    "next/dist/compiled/react-dom/cjs/react-dom-test-utils.production.js"
  ), tryResolve("next/dist/compiled/react-dom") && (addScopedAlias(baseConfig, "react-dom$", "next/dist/compiled/react-dom"), addScopedAlias(baseConfig, "react-dom/client", "next/dist/compiled/react-dom/client"), addScopedAlias(baseConfig, "react-dom/server", "next/dist/compiled/react-dom/server")), await setupRuntimeConfig(baseConfig, nextConfig), nextConfig;
}, setupRuntimeConfig = async (baseConfig, nextConfig) => {
  let definePluginConfig = {};
  isNext16orNewer || (definePluginConfig["process.env.__NEXT_RUNTIME_CONFIG"] = JSON.stringify({
    serverRuntimeConfig: {},
    publicRuntimeConfig: nextConfig.publicRuntimeConfig
  }));
  let newNextLinkBehavior = nextConfig.experimental?.newNextLinkBehavior;
  definePluginConfig["process.env.__NEXT_NEW_LINK_BEHAVIOR"] = newNextLinkBehavior, baseConfig.plugins?.push(new (await import("webpack")).default.DefinePlugin(definePluginConfig));
};

// src/font/babel/helpers.ts
function convertNodeToJSON(types, node) {
  if (types.isBooleanLiteral(node) || types.isStringLiteral(node) || types.isNumericLiteral(node))
    return node.value;
  if (!(node.name === "undefined" && !node.value))
    return types.isNullLiteral(node) ? null : types.isObjectExpression(node) ? computeProps(types, node.properties) : types.isArrayExpression(node) ? node.elements.reduce(
      (acc, element) => [
        ...acc,
        ...element?.type === "SpreadElement" ? convertNodeToJSON(types, element.argument) : [convertNodeToJSON(types, element)]
      ],
      []
    ) : {};
}
function computeProps(types, props) {
  return props.reduce((acc, prop) => {
    if (prop.type === "SpreadElement")
      return {
        ...acc,
        ...convertNodeToJSON(types, prop.argument)
      };
    if (prop.type !== "ObjectMethod") {
      let val = convertNodeToJSON(types, prop.value);
      if (val !== void 0 && types.isIdentifier(prop.key))
        return {
          ...acc,
          [prop.key.name]: val
        };
    }
    return acc;
  }, {});
}
function isDefined(value) {
  return value !== void 0;
}
function removeTransformedVariableDeclarations(path, types, metas) {
  path.parentPath.traverse({
    ExportNamedDeclaration(declaratorPath) {
      declaratorPath.parentPath?.isProgram() && metas.forEach((meta) => {
        types.isVariableDeclaration(declaratorPath.node.declaration) && declaratorPath.node.declaration.declarations.length === 1 && types.isVariableDeclarator(declaratorPath.node.declaration.declarations[0]) && types.isIdentifier(declaratorPath.node.declaration.declarations[0].id) && meta.identifierName === declaratorPath.node.declaration.declarations[0].id.name && declaratorPath.replaceWith(
          types.exportNamedDeclaration(null, [
            types.exportSpecifier(
              types.identifier(meta.identifierName),
              types.identifier(meta.identifierName)
            )
          ])
        );
      });
    },
    VariableDeclarator(declaratorPath) {
      declaratorPath.parentPath.parentPath?.isProgram() && metas.some(
        (meta) => types.isIdentifier(declaratorPath.node.id) && meta.identifierName === declaratorPath.node.id.name
      ) && declaratorPath.remove();
    }
  });
}
function replaceImportWithParameterImport(path, types, source, metas, filename) {
  path.replaceWithMultiple([
    ...metas.map((meta) => types.importDeclaration(
      [types.importDefaultSpecifier(types.identifier(meta.identifierName))],
      types.stringLiteral(
        `storybook-nextjs-font-loader?${JSON.stringify({
          source: source.value,
          props: meta.properties,
          fontFamily: meta.functionName,
          filename
        })}!${source.value}`
      )
    ))
  ]);
}
function getVariableMetasBySpecifier(program, types, specifier) {
  return program.node.body.map((statement) => {
    if (!types.isVariableDeclaration(statement) && !types.isExportNamedDeclaration(statement))
      return;
    let exportedNamedDeclaration = !types.isVariableDeclaration(statement) && types.isVariableDeclaration(statement.declaration) && statement.declaration.declarations.length === 1 ? statement.declaration.declarations[0] : null, declaration = types.isVariableDeclaration(statement) ? statement.declarations[0] : exportedNamedDeclaration;
    if (!declaration || !types.isIdentifier(declaration.id) || !types.isCallExpression(declaration.init) || !types.isIdentifier(declaration.init.callee) || (specifier.type !== "ImportSpecifier" || specifier.imported.type !== "Identifier" || declaration.init.callee.name !== specifier.imported.name && declaration.init.callee.name !== specifier.local.name) && (specifier.type !== "ImportDefaultSpecifier" || declaration.init.callee.name !== specifier.local.name))
      return;
    let options = declaration.init.arguments[0];
    if (!types.isObjectExpression(options))
      throw program.buildCodeFrameError(
        "Please pass an options object to the call expression of next/font functions"
      );
    options.properties.forEach((property) => {
      if (types.isSpreadElement(property))
        throw program.buildCodeFrameError(
          "Please do not use spread elements in the options object in next/font function calls"
        );
    });
    let identifierName = declaration.id.name, properties = convertNodeToJSON(types, options), functionName = declaration.init.callee.name;
    return specifier.type === "ImportSpecifier" && specifier.imported && specifier.imported.type === "Identifier" && declaration.init.callee.name !== specifier.imported.name && (functionName = specifier.imported.name), { identifierName, properties, functionName };
  }).filter(isDefined);
}

// src/font/babel/index.ts
function TransformFontImports({ types }) {
  return {
    name: "storybook-nextjs-font-imports",
    visitor: {
      ImportDeclaration(path, state) {
        let { node } = path, { source } = node, { filename = "" } = state;
        if (source.value === "next/font/local" || source.value === "@next/font/local") {
          let { specifiers } = node, specifier = specifiers[0];
          if (!path.parentPath.isProgram())
            return;
          let program = path.parentPath, variableMetas = getVariableMetasBySpecifier(program, types, specifier);
          removeTransformedVariableDeclarations(path, types, variableMetas), replaceImportWithParameterImport(path, types, source, variableMetas, filename);
        }
        if (source.value === "next/font/google" || source.value === "@next/font/google") {
          let { specifiers } = node, variableMetas = specifiers.flatMap((specifier) => {
            if (!path.parentPath.isProgram())
              return [];
            let program = path.parentPath;
            return getVariableMetasBySpecifier(program, types, specifier);
          }).filter(isDefined);
          removeTransformedVariableDeclarations(path, types, variableMetas), replaceImportWithParameterImport(path, types, source, variableMetas, filename);
        }
      }
    }
  };
}

// src/preset.ts
var addons = [
  fileURLToPath2(import.meta.resolve("@storybook/preset-react-webpack"))
], core = async (config, options) => {
  let framework = await options.presets.apply("framework");
  return await configureConfig({
    // Pass in a dummy webpack config object for now, since we don't want to
    // modify the real one yet. We pass in the real one in webpackFinal.
    baseConfig: {},
    nextConfigPath: typeof framework == "string" ? void 0 : framework.options.nextConfigPath
  }), {
    ...config,
    builder: {
      name: fileURLToPath2(import.meta.resolve("@storybook/builder-webpack5")),
      options: {
        ...typeof framework == "string" ? {} : framework.options.builder || {}
      }
    },
    renderer: fileURLToPath2(import.meta.resolve("@storybook/react/preset"))
  };
}, previewAnnotations = (entry = []) => {
  let annotations = [...entry, fileURLToPath2(import.meta.resolve("@storybook/nextjs/preview"))];
  return isNextVersionGte("16.0.0") || annotations.push(fileURLToPath2(import.meta.resolve("@storybook/nextjs/config/preview"))), annotations;
}, babel = async (baseConfig) => {
  let options = loadPartialConfig({
    ...baseConfig,
    filename: `${getProjectRoot()}/__fake__.js`
  })?.options, isPresetConfigItem = (preset) => typeof preset == "object" && preset !== null && "file" in preset, isNextBabelConfig = (preset) => Array.isArray(preset) && preset[0] === "next/babel" || preset === "next/babel" || isPresetConfigItem(preset) && preset.file?.request === "next/babel", hasNextBabelConfig = options?.presets?.find(isNextBabelConfig), presets = options?.presets?.filter(
    (preset) => !(isPresetConfigItem(preset) && preset.file?.request === fileURLToPath2(import.meta.resolve("@babel/preset-react")) || isNextBabelConfig(preset))
  ) ?? [];
  hasNextBabelConfig ? Array.isArray(hasNextBabelConfig) && hasNextBabelConfig[1] ? presets.push([preset_default, hasNextBabelConfig[1]]) : isPresetConfigItem(hasNextBabelConfig) && hasNextBabelConfig.file?.request === "next/babel" ? presets.push([preset_default, hasNextBabelConfig.options]) : presets.push(preset_default) : presets.push(preset_default);
  let plugins = [...options?.plugins ?? [], TransformFontImports];
  return {
    ...options,
    plugins,
    presets,
    babelrc: !1,
    configFile: !1,
    overrides: [
      ...options?.overrides ?? [],
      // We need to re-apply the default storybook babel override from:
      // https://github.com/storybookjs/storybook/blob/next/code/core/src/core-server/presets/common-preset.ts
      // Because it get lost in the loadPartialConfig call above.
      // See https://github.com/storybookjs/storybook/issues/28467
      {
        include: /(story|stories)\.[cm]?[jt]sx?$/,
        presets: [
          [
            "next/dist/compiled/babel/preset-env",
            {
              bugfixes: !0,
              targets: {
                chrome: 100,
                safari: 15,
                firefox: 91
              }
            }
          ]
        ]
      }
    ]
  };
}, webpackFinal = async (baseConfig, options) => {
  let { nextConfigPath } = await options.presets.apply("frameworkOptions"), nextConfig = await configureConfig({
    baseConfig,
    nextConfigPath
  }), { configureNextFont } = await import("./_node-chunks/configureNextFont-MD3NORSC.js"), { configureRuntimeNextjsVersionResolution, getNextjsVersion } = await import("./_node-chunks/utils-43P2NJXZ.js"), { configureImports } = await import("./_node-chunks/webpack-6JSKABBO.js"), { configureCss } = await import("./_node-chunks/webpack-EMK6SWIY.js"), { configureImages } = await import("./_node-chunks/webpack-7I3B6YHG.js"), { configureStyledJsx } = await import("./_node-chunks/webpack-XZA765NW.js"), { configureNodePolyfills } = await import("./_node-chunks/webpack-ATD3NLEM.js"), { configureAliases } = await import("./_node-chunks/webpack-FKYSP2MQ.js"), { configureFastRefresh } = await import("./_node-chunks/webpack-3KR2MQYI.js"), { configureRSC } = await import("./_node-chunks/webpack-SED5JVQ4.js"), { configureSWCLoader } = await import("./_node-chunks/loader-FLICOCFJ.js"), { configureBabelLoader } = await import("./_node-chunks/loader-RQPJEVRW.js"), babelRCPath = join(getProjectRoot(), ".babelrc"), babelConfigPath = join(getProjectRoot(), "babel.config.js"), hasBabelConfig = existsSync(babelRCPath) || existsSync(babelConfigPath), nextjsVersion = getNextjsVersion(), isDevelopment = options.configType !== "PRODUCTION", useSWC = semver.gte(nextjsVersion, "14.0.0") && (nextConfig.experimental?.forceSwcTransforms || !hasBabelConfig);
  return configureNextFont(baseConfig, useSWC), configureRuntimeNextjsVersionResolution(baseConfig), configureImports({ baseConfig, configDir: options.configDir }), configureCss(baseConfig, nextConfig), configureImages(baseConfig, nextConfig), configureStyledJsx(baseConfig), configureNodePolyfills(baseConfig), configureAliases(baseConfig), isDevelopment && configureFastRefresh(baseConfig), options.features?.experimentalRSC && configureRSC(baseConfig), useSWC ? (logger.info("Using SWC as compiler"), await configureSWCLoader(baseConfig, options, nextConfig)) : (logger.info("Using Babel as compiler"), await configureBabelLoader(baseConfig, options, nextConfig)), baseConfig;
};
export {
  addons,
  babel,
  core,
  previewAnnotations,
  webpackFinal
};
