// @storybook/nextjs/dist/swc/next-swc-loader-patch.js
import CJS_COMPAT_NODE_URL_ii3a4tu75d7 from 'node:url';
import CJS_COMPAT_NODE_PATH_ii3a4tu75d7 from 'node:path';
import CJS_COMPAT_NODE_MODULE_ii3a4tu75d7 from "node:module";

var __filename = CJS_COMPAT_NODE_URL_ii3a4tu75d7.fileURLToPath(import.meta.url);
var __dirname = CJS_COMPAT_NODE_PATH_ii3a4tu75d7.dirname(__filename);
var require = CJS_COMPAT_NODE_MODULE_ii3a4tu75d7.createRequire(import.meta.url);

// ------------------------------------------------------------
// end of CJS compatibility banner, injected by Storybook's esbuild configuration
// ------------------------------------------------------------

// src/swc/next-swc-loader-patch.ts
import { isAbsolute, relative } from "node:path";
import * as nextSwcUtils from "next/dist/build/swc/index.js";
import { getLoaderSWCOptions } from "next/dist/build/swc/options.js";
var mockCurrentTraceSpan = {
  traceChild: (name) => mockCurrentTraceSpan,
  traceAsyncFn: async (fn) => fn()
};
async function loaderTransform(parentTrace, source, inputSourceMap) {
  let filename = this.resourcePath, loaderOptions = this.getOptions() || {}, {
    isServer,
    rootDir,
    pagesDir,
    appDir,
    hasReactRefresh,
    nextConfig,
    jsConfig,
    supportedBrowsers,
    swcCacheDir,
    serverComponents,
    isReactServerLayer
  } = loaderOptions, isPageFile = filename.startsWith(pagesDir), relativeFilePathFromRoot = relative(rootDir, filename), programmaticOptions = {
    ...getLoaderSWCOptions({
      pagesDir,
      appDir,
      filename,
      isServer,
      isPageFile,
      development: this.mode === "development",
      hasReactRefresh,
      modularizeImports: nextConfig?.modularizeImports,
      optimizePackageImports: nextConfig?.experimental?.optimizePackageImports,
      swcPlugins: nextConfig?.experimental?.swcPlugins,
      compilerOptions: nextConfig?.compiler,
      optimizeServerReact: nextConfig?.experimental?.optimizeServerReact,
      jsConfig,
      supportedBrowsers,
      swcCacheDir,
      relativeFilePathFromRoot,
      serverComponents,
      // @ts-expect-error Relevant for Next.js < 14.1
      // TODO: Remove this when Next.js < 14.1 is no longer supported
      isReactServerLayer
    }),
    filename,
    inputSourceMap: inputSourceMap && typeof inputSourceMap == "object" ? JSON.stringify(inputSourceMap) : void 0,
    // Set the default sourcemap behavior based on Webpack's mapping flag,
    sourceMaps: this.sourceMap,
    inlineSourcesContent: this.sourceMap,
    // Ensure that Webpack will get a full absolute path in the sourcemap
    // so that it can properly map the module back to its internal cached
    // modules.
    sourceFileName: filename
  };
  return programmaticOptions.env.bugfixes = !0, programmaticOptions.inputSourceMap || delete programmaticOptions.inputSourceMap, this.mode && programmaticOptions.jsc && programmaticOptions.jsc.transform && programmaticOptions.jsc.transform.react && !Object.prototype.hasOwnProperty.call(programmaticOptions.jsc.transform.react, "development") && (programmaticOptions.jsc.transform.react.development = this.mode === "development"), parentTrace.traceChild("next-swc-transform").traceAsyncFn(
    () => nextSwcUtils.transform(source, programmaticOptions).then((output) => {
      if (output.eliminatedPackages && this.eliminatedPackages)
        for (let pkg of JSON.parse(output.eliminatedPackages))
          this.eliminatedPackages.add(pkg);
      return [output.code, output.map ? JSON.parse(output.map) : void 0];
    })
  );
}
var EXCLUDED_PATHS = /[\\/](cache[\\/][^\\/]+\.zip[\\/]node_modules|__virtual__)[\\/]/g;
function pitch() {
  let callback = this.async();
  (async () => {
    let isWasm2 = !1;
    if (nextSwcUtils.isWasm ? isWasm2 = await nextSwcUtils.isWasm() : nextSwcUtils.getBindingsSync && (await nextSwcUtils.loadBindings(), isWasm2 = nextSwcUtils.getBindingsSync().isWasm), // TODO: Evaluate if this is correct after removing pnp compatibility code in SB11
    // TODO: investigate swc file reading in PnP mode?
    !process.versions.pnp && !EXCLUDED_PATHS.test(this.resourcePath) && this.loaders.length - 1 === this.loaderIndex && isAbsolute(this.resourcePath) && !isWasm2) {
      let loaderSpan = mockCurrentTraceSpan.traceChild("next-swc-loader");
      return this.addDependency(this.resourcePath), loaderSpan.traceAsyncFn(() => loaderTransform.call(this, loaderSpan));
    }
    return null;
  })().then((r) => r ? callback(null, ...r) : (callback(), null), callback);
}
function swcLoader(inputSource, inputSourceMap) {
  let loaderSpan = mockCurrentTraceSpan.traceChild("next-swc-loader"), callback = this.async();
  loaderSpan.traceAsyncFn(() => loaderTransform.call(this, loaderSpan, inputSource, inputSourceMap)).then(
    ([transformedSource, outputSourceMap]) => {
      callback(null, transformedSource, outputSourceMap || inputSourceMap);
    },
    (err) => {
      callback(err);
    }
  );
}
var raw = !0;
export {
  swcLoader as default,
  pitch,
  raw
};
