// vite-plugin-storybook-nextjs/dist/plugins/next-mocks/alias/cache/index.js
import { fn } from 'storybook/test';

// src/plugins/next-mocks/alias/cache/index.ts
var revalidatePath = fn().mockName(
  "next/cache::revalidatePath"
);
var revalidateTag = fn().mockName(
  "next/cache::revalidateTag"
);
var updateTag = fn().mockName("next/cache::updateTag");
var unstable_cache = fn().mockName("next/cache::unstable_cache").mockImplementation((cb) => cb);
var unstable_noStore = fn().mockName(
  "next/cache::unstable_noStore"
);
var refresh = fn().mockName("next/cache::refresh");
var cacheLife = fn().mockName("next/cache::cacheLife");
var cacheTag = fn().mockName("next/cache::cacheTag");
var unstable_cacheLife = fn().mockName("next/cache::unstable_cacheLife").mockImplementation((...args) => cacheLife(...args));
var unstable_cacheTag = fn().mockName("next/cache::unstable_cacheTag").mockImplementation((...args) => cacheTag(...args));
var cacheExports = {
  unstable_cache,
  revalidateTag,
  revalidatePath,
  updateTag,
  refresh,
  unstable_noStore,
  cacheLife,
  cacheTag,
  unstable_cacheLife,
  unstable_cacheTag
};
var cache_default = cacheExports;

export { cacheLife, cacheTag, cache_default as default, refresh, revalidatePath, revalidateTag, unstable_cache, unstable_cacheLife, unstable_cacheTag, unstable_noStore, updateTag };
