// vite-plugin-storybook-nextjs/dist/plugins/next-mocks/alias/cache/index.cjs
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var test = require('storybook/test');

// src/plugins/next-mocks/alias/cache/index.ts
var revalidatePath = test.fn().mockName(
  "next/cache::revalidatePath"
);
var revalidateTag = test.fn().mockName(
  "next/cache::revalidateTag"
);
var updateTag = test.fn().mockName("next/cache::updateTag");
var unstable_cache = test.fn().mockName("next/cache::unstable_cache").mockImplementation((cb) => cb);
var unstable_noStore = test.fn().mockName(
  "next/cache::unstable_noStore"
);
var refresh = test.fn().mockName("next/cache::refresh");
var cacheLife = test.fn().mockName("next/cache::cacheLife");
var cacheTag = test.fn().mockName("next/cache::cacheTag");
var unstable_cacheLife = test.fn().mockName("next/cache::unstable_cacheLife").mockImplementation((...args) => cacheLife(...args));
var unstable_cacheTag = test.fn().mockName("next/cache::unstable_cacheTag").mockImplementation((...args) => cacheTag(...args));
var cacheExports = {
  unstable_cache,
  revalidateTag,
  revalidatePath,
  updateTag,
  refresh,
  unstable_noStore,
  cacheLife,
  cacheTag,
  unstable_cacheLife,
  unstable_cacheTag
};
var cache_default = cacheExports;

exports.cacheLife = cacheLife;
exports.cacheTag = cacheTag;
exports.default = cache_default;
exports.refresh = refresh;
exports.revalidatePath = revalidatePath;
exports.revalidateTag = revalidateTag;
exports.unstable_cache = unstable_cache;
exports.unstable_cacheLife = unstable_cacheLife;
exports.unstable_cacheTag = unstable_cacheTag;
exports.unstable_noStore = unstable_noStore;
exports.updateTag = updateTag;
