// vite-plugin-storybook-nextjs/dist/plugins/next-mocks/alias/headers/headers.cjs
'use strict';

var test = require('storybook/test');
var headers_js = require('next/dist/server/web/spec-extension/adapters/headers.js');

// src/plugins/next-mocks/alias/headers/headers.ts
var HeadersAdapterMock = class extends headers_js.HeadersAdapter {
  constructor() {
    super({});
    this.append = test.fn(
      super.append.bind(this)
    ).mockName("next/headers::headers().append");
    this.delete = test.fn(super.delete.bind(this)).mockName(
      "next/headers::headers().delete"
    );
    this.get = test.fn(
      super.get.bind(this)
    ).mockName("next/headers::headers().get");
    this.has = test.fn(super.has.bind(this)).mockName(
      "next/headers::headers().has"
    );
    this.set = test.fn(
      super.set.bind(this)
    ).mockName("next/headers::headers().set");
    this.forEach = test.fn(super.forEach.bind(this)).mockName("next/headers::headers().forEach");
    this.entries = test.fn(
      super.entries.bind(this)
    ).mockName("next/headers::headers().entries");
    this.keys = test.fn(
      super.keys.bind(this)
    ).mockName("next/headers::headers().keys");
    this.values = test.fn(
      super.values.bind(this)
    ).mockName("next/headers::headers().values");
  }
};
var headersAdapterMock;
var headers = () => {
  if (!headersAdapterMock) headersAdapterMock = new HeadersAdapterMock();
  return headersAdapterMock;
};
headers.mockRestore = () => {
  headersAdapterMock = new HeadersAdapterMock();
};

exports.headers = headers;
