// @storybook/nextjs/dist/_node-chunks/chunk-HTZOSU3S.js
import CJS_COMPAT_NODE_URL_ii3a4tu75d7 from 'node:url';
import CJS_COMPAT_NODE_PATH_ii3a4tu75d7 from 'node:path';
import CJS_COMPAT_NODE_MODULE_ii3a4tu75d7 from "node:module";

var __filename = CJS_COMPAT_NODE_URL_ii3a4tu75d7.fileURLToPath(import.meta.url);
var __dirname = CJS_COMPAT_NODE_PATH_ii3a4tu75d7.dirname(__filename);
var require = CJS_COMPAT_NODE_MODULE_ii3a4tu75d7.createRequire(import.meta.url);

// ------------------------------------------------------------
// end of CJS compatibility banner, injected by Storybook's esbuild configuration
// ------------------------------------------------------------
import {
  resolvePackageDir
} from "./chunk-4TF7YMRH.js";

// src/utils.ts
import { readFileSync } from "node:fs";
import { dirname, join, sep } from "node:path";
import { fileURLToPath } from "node:url";
import { getProjectRoot } from "storybook/internal/common";
import { WebpackDefinePlugin } from "@storybook/builder-webpack5";
import { PHASE_DEVELOPMENT_SERVER } from "next/constants.js";
import nextJsLoadConfigModule from "next/dist/server/config.js";
import semver from "semver";
var configureRuntimeNextjsVersionResolution = (baseConfig) => {
  baseConfig.plugins?.push(
    new WebpackDefinePlugin({
      "process.env.__NEXT_VERSION": JSON.stringify(getNextjsVersion())
    })
  );
}, getNextjsVersion = () => JSON.parse(readFileSync(join(resolvePackageDir("next"), "package.json"), "utf8")).version, isNextVersionGte = (version) => {
  let currentVersion = getNextjsVersion(), coercedVersion = semver.coerce(currentVersion);
  return coercedVersion ? semver.gte(coercedVersion, version) : !1;
}, resolveNextConfig = async ({
  nextConfigPath
}) => {
  let dir = nextConfigPath ? dirname(nextConfigPath) : getProjectRoot(), loadConfig = nextJsLoadConfigModule.default ?? nextJsLoadConfigModule, nextPrivateRenderWorker = process.env.__NEXT_PRIVATE_RENDER_WORKER;
  process.env.__NEXT_PRIVATE_RENDER_WORKER = "defined";
  let config = loadConfig(PHASE_DEVELOPMENT_SERVER, dir, void 0);
  return typeof nextPrivateRenderWorker > "u" ? delete process.env.__NEXT_PRIVATE_RENDER_WORKER : process.env.__NEXT_PRIVATE_RENDER_WORKER = nextPrivateRenderWorker, config;
};
function setAlias(baseConfig, name, alias) {
  baseConfig.resolve ??= {}, baseConfig.resolve.alias ??= {};
  let aliasConfig = baseConfig.resolve.alias;
  Array.isArray(aliasConfig) ? aliasConfig.push({
    name,
    alias
  }) : aliasConfig[name] = alias;
}
var addScopedAlias = (baseConfig, name, alias) => {
  let scopedAlias = scopedResolve(`${alias ?? name}`);
  setAlias(baseConfig, name, scopedAlias);
}, scopedResolve = (id) => {
  let scopedModulePath = fileURLToPath(import.meta.resolve(id)), idWithNativePathSep = id.replace(/\//g, sep);
  if (scopedModulePath.endsWith(idWithNativePathSep))
    return scopedModulePath;
  let beginningOfMainScriptPath = scopedModulePath.lastIndexOf(idWithNativePathSep) + id.length;
  return scopedModulePath.substring(0, beginningOfMainScriptPath);
};
function getNodeModulesExcludeRegex(transpilePackages) {
  if (!transpilePackages || transpilePackages.length === 0)
    return /node_modules/;
  let escaped = transpilePackages.map((pkg) => pkg.replace(/[.*+?^${}()|[\]\\]/g, "\\$&")).join("|");
  return new RegExp(`node_modules/(?!(${escaped})/)`);
}

export {
  configureRuntimeNextjsVersionResolution,
  getNextjsVersion,
  isNextVersionGte,
  resolveNextConfig,
  setAlias,
  addScopedAlias,
  scopedResolve,
  getNodeModulesExcludeRegex
};
