// @storybook/nextjs/dist/_node-chunks/loader-FLICOCFJ.js
import CJS_COMPAT_NODE_URL_ii3a4tu75d7 from 'node:url';
import CJS_COMPAT_NODE_PATH_ii3a4tu75d7 from 'node:path';
import CJS_COMPAT_NODE_MODULE_ii3a4tu75d7 from "node:module";

var __filename = CJS_COMPAT_NODE_URL_ii3a4tu75d7.fileURLToPath(import.meta.url);
var __dirname = CJS_COMPAT_NODE_PATH_ii3a4tu75d7.dirname(__filename);
var require = CJS_COMPAT_NODE_MODULE_ii3a4tu75d7.createRequire(import.meta.url);

// ------------------------------------------------------------
// end of CJS compatibility banner, injected by Storybook's esbuild configuration
// ------------------------------------------------------------
import {
  getNodeModulesExcludeRegex
} from "./chunk-HTZOSU3S.js";
import "./chunk-4TF7YMRH.js";

// src/swc/loader.ts
import { join } from "node:path";
import { getProjectRoot } from "storybook/internal/common";
import { getVirtualModules } from "@storybook/builder-webpack5";
import nextJSLoadConfigModule from "next/dist/build/load-jsconfig.js";
var configureSWCLoader = async (baseConfig, options, nextConfig) => {
  let isDevelopment = options.configType !== "PRODUCTION", { virtualModules } = await getVirtualModules(options), projectRoot = getProjectRoot(), loadJsConfig = nextJSLoadConfigModule.default ?? nextJSLoadConfigModule, { jsConfig } = await loadJsConfig(projectRoot, nextConfig), rawRule = baseConfig.module?.rules?.find(
    (rule) => typeof rule == "object" && rule?.resourceQuery?.toString() === "/raw/"
  );
  rawRule && typeof rawRule == "object" && (rawRule.exclude = /^__barrel_optimize__/);
  let transpilePackages = nextConfig.transpilePackages ?? [];
  baseConfig.module?.rules?.push({
    test: /\.((c|m)?(j|t)sx?)$/,
    include: [projectRoot],
    exclude: [getNodeModulesExcludeRegex(transpilePackages), ...Object.keys(virtualModules)],
    use: {
      // we use our own patch because we need to remove tracing from the original code
      // which is not possible otherwise
      loader: "@storybook/nextjs/next-swc-loader-patch",
      options: {
        isServer: !1,
        rootDir: projectRoot,
        pagesDir: `${projectRoot}/pages`,
        appDir: `${projectRoot}/apps`,
        hasReactRefresh: isDevelopment,
        jsConfig,
        nextConfig,
        supportedBrowsers: await getSupportedBrowsers(projectRoot, isDevelopment),
        swcCacheDir: join(projectRoot, nextConfig?.distDir ?? ".next", "cache", "swc"),
        bundleTarget: "default"
      }
    }
  });
};
async function getSupportedBrowsers(projectRoot, isDevelopment) {
  try {
    return (await import("next/dist/build/get-supported-browsers.js")).getSupportedBrowsers(
      projectRoot,
      isDevelopment
    );
  } catch {
    return (await import("next/dist/build/utils.js")).getSupportedBrowsers(
      projectRoot,
      isDevelopment
    );
  }
}
export {
  configureSWCLoader
};
