// @storybook/nextjs/dist/_node-chunks/loader-RQPJEVRW.js
import CJS_COMPAT_NODE_URL_ii3a4tu75d7 from 'node:url';
import CJS_COMPAT_NODE_PATH_ii3a4tu75d7 from 'node:path';
import CJS_COMPAT_NODE_MODULE_ii3a4tu75d7 from "node:module";

var __filename = CJS_COMPAT_NODE_URL_ii3a4tu75d7.fileURLToPath(import.meta.url);
var __dirname = CJS_COMPAT_NODE_PATH_ii3a4tu75d7.dirname(__filename);
var require = CJS_COMPAT_NODE_MODULE_ii3a4tu75d7.createRequire(import.meta.url);

// ------------------------------------------------------------
// end of CJS compatibility banner, injected by Storybook's esbuild configuration
// ------------------------------------------------------------
import {
  getNodeModulesExcludeRegex
} from "./chunk-HTZOSU3S.js";
import "./chunk-4TF7YMRH.js";

// src/babel/loader.ts
import { fileURLToPath } from "node:url";
import { getProjectRoot, resolvePathInStorybookCache } from "storybook/internal/common";
import { getVirtualModules } from "@storybook/builder-webpack5";
var configureBabelLoader = async (baseConfig, options, nextConfig) => {
  let { virtualModules } = await getVirtualModules(options), babelOptions = await options.presets.apply("babel", {}, options), typescriptOptions = await options.presets.apply("typescript", {}, options);
  baseConfig.module.rules = [
    ...baseConfig.module.rules,
    {
      test: typescriptOptions.skipCompiler ? /\.((c|m)?jsx?)$/ : /\.((c|m)?(j|t)sx?)$/,
      use: [
        {
          loader: fileURLToPath(import.meta.resolve("babel-loader")),
          options: {
            cacheDirectory: resolvePathInStorybookCache("babel"),
            ...babelOptions
          }
        }
      ],
      include: [getProjectRoot()],
      exclude: [
        getNodeModulesExcludeRegex(nextConfig.transpilePackages ?? []),
        ...Object.keys(virtualModules)
      ]
    }
  ];
};
export {
  configureBabelLoader
};
