// @storybook/nextjs/dist/_node-chunks/webpack-7I3B6YHG.js
import CJS_COMPAT_NODE_URL_ii3a4tu75d7 from 'node:url';
import CJS_COMPAT_NODE_PATH_ii3a4tu75d7 from 'node:path';
import CJS_COMPAT_NODE_MODULE_ii3a4tu75d7 from "node:module";

var __filename = CJS_COMPAT_NODE_URL_ii3a4tu75d7.fileURLToPath(import.meta.url);
var __dirname = CJS_COMPAT_NODE_PATH_ii3a4tu75d7.dirname(__filename);
var require = CJS_COMPAT_NODE_MODULE_ii3a4tu75d7.createRequire(import.meta.url);

// ------------------------------------------------------------
// end of CJS compatibility banner, injected by Storybook's esbuild configuration
// ------------------------------------------------------------
import {
  getNextjsVersion
} from "./chunk-HTZOSU3S.js";
import "./chunk-4TF7YMRH.js";

// src/images/webpack.ts
import { fileURLToPath } from "node:url";
import semver from "semver";
var configureImages = (baseConfig, nextConfig) => {
  configureStaticImageImport(baseConfig, nextConfig), configureImageDefaults(baseConfig);
}, fallbackFilename = "static/media/[path][name][ext]", configureImageDefaults = (baseConfig) => {
  let version = getNextjsVersion(), resolve = baseConfig.resolve ?? {};
  resolve.alias = {
    ...resolve.alias,
    "sb-original/next/image": fileURLToPath(import.meta.resolve("next/image")),
    "next/image": fileURLToPath(import.meta.resolve("@storybook/nextjs/images/next-image"))
  }, semver.satisfies(version, ">=13.0.0") && (resolve.alias = {
    ...resolve.alias,
    "sb-original/next/legacy/image": fileURLToPath(import.meta.resolve("next/legacy/image")),
    "next/legacy/image": fileURLToPath(
      import.meta.resolve("@storybook/nextjs/images/next-legacy-image")
    )
  });
}, configureStaticImageImport = (baseConfig, nextConfig) => {
  let rules = baseConfig.module?.rules, assetRule = rules?.find(
    (rule) => rule && typeof rule != "string" && rule.test instanceof RegExp && rule.test.test("test.jpg")
  );
  assetRule && (assetRule.test = /\.(apng|eot|otf|ttf|woff|woff2|cur|ani|pdf)(\?.*)?$/, rules?.push({
    test: /\.(png|jpg|jpeg|gif|webp|avif|ico|bmp|svg)$/i,
    issuer: { not: /\.(css|scss|sass)$/ },
    use: [
      {
        loader: fileURLToPath(import.meta.resolve("@storybook/nextjs/next-image-loader-stub")),
        options: {
          filename: assetRule.generator?.filename ?? fallbackFilename,
          nextConfig
        }
      }
    ]
  }), rules?.push({
    test: /\.(png|jpg|jpeg|gif|webp|avif|ico|bmp|svg)$/i,
    issuer: /\.(css|scss|sass)$/,
    type: "asset/resource",
    generator: {
      filename: assetRule.generator?.filename ?? fallbackFilename
    }
  }));
};
export {
  configureImages
};
