// @storybook/nextjs/dist/_node-chunks/webpack-EMK6SWIY.js
import CJS_COMPAT_NODE_URL_ii3a4tu75d7 from 'node:url';
import CJS_COMPAT_NODE_PATH_ii3a4tu75d7 from 'node:path';
import CJS_COMPAT_NODE_MODULE_ii3a4tu75d7 from "node:module";

var __filename = CJS_COMPAT_NODE_URL_ii3a4tu75d7.fileURLToPath(import.meta.url);
var __dirname = CJS_COMPAT_NODE_PATH_ii3a4tu75d7.dirname(__filename);
var require = CJS_COMPAT_NODE_MODULE_ii3a4tu75d7.createRequire(import.meta.url);

// ------------------------------------------------------------
// end of CJS compatibility banner, injected by Storybook's esbuild configuration
// ------------------------------------------------------------
import {
  resolvePackageDir
} from "./chunk-4TF7YMRH.js";

// src/css/webpack.ts
import { readFileSync } from "node:fs";
import { join } from "node:path";
import { fileURLToPath } from "node:url";
import { cssFileResolve } from "next/dist/build/webpack/config/blocks/css/loaders/file-resolve.js";
import { getCssModuleLocalIdent } from "next/dist/build/webpack/config/blocks/css/loaders/getCssModuleLocalIdent.js";
import semver from "semver";
var configureCss = (baseConfig, nextConfig) => {
  let rules = baseConfig.module?.rules;
  rules?.forEach((rule, i) => {
    rule && typeof rule != "string" && rule.test instanceof RegExp && rule.test.test("test.css") && (rules[i] = {
      test: /\.css$/,
      use: [
        fileURLToPath(import.meta.resolve("style-loader")),
        {
          loader: fileURLToPath(import.meta.resolve("css-loader")),
          options: {
            importLoaders: 1,
            ...getImportAndUrlCssLoaderOptions(nextConfig),
            modules: {
              auto: !0,
              getLocalIdent: getCssModuleLocalIdent
            }
          }
        },
        fileURLToPath(import.meta.resolve("postcss-loader"))
      ],
      // We transform the "target.css" files from next.js into Javascript
      // for Next.js to support fonts, so it should be ignored by the css-loader.
      exclude: /next(\\|\/|\\\\).*(\\|\/|\\\\)target\.css$/
    });
  }), rules?.push({
    test: /\.(scss|sass)$/,
    use: [
      fileURLToPath(import.meta.resolve("style-loader")),
      {
        loader: fileURLToPath(import.meta.resolve("css-loader")),
        options: {
          importLoaders: 3,
          ...getImportAndUrlCssLoaderOptions(nextConfig),
          modules: { auto: !0, getLocalIdent: getCssModuleLocalIdent }
        }
      },
      fileURLToPath(import.meta.resolve("postcss-loader")),
      fileURLToPath(import.meta.resolve("resolve-url-loader")),
      {
        loader: fileURLToPath(import.meta.resolve("sass-loader")),
        options: {
          sourceMap: !0,
          sassOptions: nextConfig.sassOptions,
          additionalData: nextConfig.sassOptions?.prependData || nextConfig.sassOptions?.additionalData
        }
      }
    ]
  });
}, getImportAndUrlCssLoaderOptions = (nextConfig) => isCssLoaderV6() ? {
  url: {
    filter: getUrlResolver(nextConfig)
  },
  import: {
    filter: getImportResolver(nextConfig)
  }
} : {
  url: getUrlResolver(nextConfig),
  import: getImportResolver(nextConfig)
}, getUrlResolver = (nextConfig) => (url, resourcePath) => cssFileResolve(url, resourcePath, nextConfig.experimental?.urlImports), getImportResolver = (nextConfig) => (url, _, resourcePath) => cssFileResolve(
  typeof url == "string" ? url : url.url,
  resourcePath,
  nextConfig.experimental?.urlImports
), isCssLoaderV6 = () => {
  try {
    let cssLoaderVersion = JSON.parse(
      readFileSync(join(resolvePackageDir("css-loader"), "package.json"), "utf8")
    ).version;
    return semver.gte(cssLoaderVersion, "6.0.0");
  } catch {
    return !1;
  }
};
export {
  configureCss
};
